playbooks/files/backup/backup.sh

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env -S sh -e -u
+#   Copyright (C) 2024 Martina Ferrari, Simó Albert i Beltran
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU Affero General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+BACKUP_FROM_DIR="${BACKUP_FROM_DIR:-/tmp/bk-snap}"
+
+get_backup_location(){
+  # Get backup location for loop device name
+  local loop_device_name=$1
+  echo $BACKUP_FROM_DIR/$(echo $loop_device_name | sed 's/--snap//')
+}
+
+get_volumes(){
+  # Get LVM volumes that are not snapshots
+  lvs --noheadings -o vg_name,lv_name,lv_attr | awk '$3 ~ /^[-oV]/ {print $1"/"$2}'
+}
+
+is_partition_table(){
+  # Exit with 0 if the parameter is a device with a partition table, 1 otherwise.
+  blkid $1 | grep -q " PTTYPE="
+}
+
+is_filesystem(){
+  # Exit with 0 if the parameter is a device with a filesystem, 1 otherwise.
+  # TODO(sim6): Detection should be improved to only return 0 if the filesystem can be
+  # mounted.
+  # TODO(sim6): Swap partition should not be detected as filesystem.
+  blkid $1 | grep -q " TYPE="
+}
+
+teardown(){
+  # Remove MBR and partition tables backup
+  lsblk -l -o TYPE,NAME | grep "^disk " | while read _ harddrive
+  do
+    rm -f $BACKUP_FROM_DIR/$harddrive.mbr
+    rm -f $BACKUP_FROM_DIR/$harddrive.sfdisk
+  done
+  # Umount boot and EFI
+  for mount_point in boot boot_efi
+  do
+    mountpoint -q $BACKUP_FROM_DIR/$mount_point &&
+      umount $BACKUP_FROM_DIR/$mount_point
+  done
+  # Umount filesystems and remove snapshots
+  get_volumes | while read volume
+  do
+    test ! -b /dev/$volume-snap &&
+      continue
+    # The created snapshot can contain a disk with partitions that are mounted or
+    # dumped. These partitions will be unmounted or dumps will be deleted.
+    # Otherwise, if the created snapshot does not contain a disk with partitions, the
+    # snapshot will be unmounted or dump will be deleted.
+    if is_partition_table /dev/$volume-snap
+    then
+      kpartx -l /dev/$volume-snap | while read loop_device _
+      do
+        findmnt /dev/mapper/$loop_device -o TARGET | while read mount_point
+        do
+	  umount /dev/mapper/$loop_device
+	  rmdir $mount_point
+	done
+	test -f $(get_backup_location $loop_device) &&
```

Comment:
These lines (and several others below) are indented with tabs, resulting in wrong alignment depending on configured tab width. Could you perhaps ensure all lines are indented with spaces?

Reply:
Apologies, I will fix it!

Reply:
Done via https://github.com/Tina-pm/ansible-playbooks/pull/20/commits/6070b90f5b02e2b9eb1b856404ece16b864f6503